components/FormikFields/PaperInputPicker/index.js

```diff
@@ -34,6 +37,65 @@ const PaperInputPicker = ({
 
   const [location, setLocation] = React.useState({ latitude: 0, longitude: 0, altitude: 0 });
   const [locationLoading, setLocationLoading] = React.useState(false);
+  const [numberedQestions, setNumberedQuestions] = React.useState({})
+  const [loopPosition, setLoopPosition] = React.useState(-1);
+  const [questionsToRepeat, setQuestionsToRepeat] = React.useState([]);
+  const [loopsAdded, setLoopsAdded] = React.useState(0);
+  const [additionalQuestions, setAdditionalQuestions] = useState([]);
+
+  useEffect(() => {
+    let fieldsNumberedJson = {}
+    // console.log(config.fields)
+    config.fields.forEach((field, index) => {
+      fieldsNumberedJson[index] = field;
+    })
+    setNumberedQuestions(fieldsNumberedJson)
+  }, [])
+
+  useEffect(() => {
+    if (fieldType === 'loop') {
+      Object.entries(numberedQestions).forEach(([key, value]) => {
+        if (value.formikKey === formikKey) {
+          setLoopPosition(key);
+          let repeatQuestions = []
+          for (let i = key - numberQuestionsToRepeat; i <  key; i++) {
+            repeatQuestions = repeatQuestions.concat(numberedQestions[i]);
+          }
+          setQuestionsToRepeat(repeatQuestions);
+        }
+      })
+    }
```

Comment:
@jamccomb92 what you can do is turn the code in this and the previous `useEffect` into two separate functions, and then put both under one `useEffect`